packages/integration-tests/index.js

```diff
@@ -656,18 +655,30 @@ async function buildWorkflow(
   const template = await getJsonS3Object(bucketName, templateKey(stackName));
 
   if (collection) {
-    template.meta.collection = await collectionsApi.getCollection({
+    const collectionsApiResponse = await collectionsApi.getCollection({
       prefix: stackName,
       collectionName: collection.name,
       collectionVersion: collection.version,
     });
+    if (collectionsApiResponse.statusCode) {
```

Comment:
Is this condition correct? Do we mean to do this:

```suggestion
    if (collectionsApiResponse.statusCode !== 200) {
```

Reply:
No - because it never equals 200.  Critically the response back from  collectionsApi.getCollection returns something like this when successful: 

url_path:"{cmrMetadata.Granule.Collection.ShortName}___{cmrMetadata.Granule.Collection.VersionId}/{substring(file.name, 0, 3)}"
updatedAt:1621617065051
sampleFileName:"MOD09GQ.A2017025.h21v00.006.2017034065104.hdf"
reportToEms:false
process:"modis"
name:"MOD09GQ_test-jk-tf4-IngestFromPdr-1621617055736"
granuleIdExtraction:"(MOD09GQ\..*)(\.hdf|\.cmr|_ndvi\.jpg)"
granuleId:"^MOD09GQ\.A[\d]{7}\.[\S]{6}\.006\.[\d]{13}$"
files:Array(5) [Object, Object, Object, …]
duplicateHandling:"replace"
createdAt:1621617065051

but on anything that throws: 

message:"No record found for {"name":"foobar","version":"006"} in jk-tf4-CollectionsTable; Function params" <snip>
error: "Not Found"
statusCode: 404


and *that* is because `getCollection`  returns ` return JSON.parse(returnedCollection.body);`, which in the case of a success is the object, but in the case of a failure from the api lambda is a body object with the return code and error info. 

This method is a bit strange, as *originally* before the API package was implemented it parsed the body unlike the other API calls.

I'm happy to walk through that if it doesn't make sense - this oddness is part of where my comment below about a refactor comes into play.

Reply:
I'll take your word for it for now